larsim/LegacyLArG4/OpFastScintillation.cxx

```diff
@@ -274,29 +279,38 @@ namespace larg4 {
         fPVS->LoadTimingsForVUVPar(fparameters,
                                    fstep_size,
                                    fmax_d,
+                                   fmin_d,
                                    fvuv_vgroup_mean,
                                    fvuv_vgroup_max,
-                                   finflexion_point_distance);
+                                   finflexion_point_distance,
+                                   fangle_bin_timing_vuv);
 
         // create vector of empty TF1s that will be replaces with the parameterisations that are generated as they are required
         // default TF1() constructor gives function with 0 dimensions, can then check numDim to qucikly see if a parameterisation has been generated
-        const size_t num_params =
-          (fmax_d - 25) /
-          fstep_size; // for d < 25cm, no parameterisaton, a delta function is used instead // TODO: unhardcode
+        const size_t num_params = (fmax_d - fmin_d) / fstep_size; // for d < fmin_d, no parameterisaton, a delta function is used instead
+        size_t num_angles = std::round(90/fangle_bin_timing_vuv);
         std::vector<TF1> VUV_timing_temp(num_params, TF1());
-        VUV_timing = VUV_timing_temp;
+        for (size_t i = 0; i < num_angles; ++i) {
+          VUV_timing.push_back(VUV_timing_temp);
+        }
```

Comment:
This can be abbreviated:
```.cpp
VUV_timing = std::vector(num_angles, std::vector(num_params, TF1{}));
```
I encourage you to follow a similar pattern for setting the vector member variables.